In [1]:
#Package
import pandas as pd
import numpy as np
import matplotlib
from matplotlib import pyplot as plt
import seaborn as sns
from matplotlib import font_manager, rc
plt.rcParams['font.family'] = 'NanumGothic'

In [2]:
import warnings
warnings.filterwarnings(action='ignore')

In [3]:
train=pd.read_csv("train.csv")
test=pd.read_csv("test.csv")

In [4]:
for col in ['임대료','임대보증금']:
    train[col] = train[col].replace('-',np.nan)
    train[col].fillna(-1,inplace=True)
    train[col] = train[col].astype('float')
    train[col] = train[col].replace(-1,np.nan)

for col in ['임대료','임대보증금']:
    test[col] = test[col].replace('-',np.nan)
    test[col].fillna(-1,inplace=True)
    test[col] = test[col].astype('float')
    test[col] = test[col].replace(-1,np.nan)

In [5]:
train.columns = ['단지코드', '총세대수', '임대건물구분', '지역', '공급유형', '전용면적', '전용면적별세대수', '공가수',
       '자격유형', '임대보증금', '임대료', '지하철역',
       '버스정류장', '단지내주차면수', '등록차량수']
test.columns = ['단지코드', '총세대수', '임대건물구분', '지역', '공급유형', '전용면적', '전용면적별세대수', '공가수',
       '자격유형', '임대보증금', '임대료', '지하철역',
       '버스정류장', '단지내주차면수']

In [6]:
data = pd.read_csv('external_data.csv',header='infer')
data.head()

기관명  광역시도코드  광역시도명  시군구코드 시군구명        단지명                  도로명주소  세대수  \
0  LH서울      11  서울특별시    110  종로구  서울특별시 종로구  서울특별시 종로구 종로31가길 25-5   66   
1  LH서울      11  서울특별시    110  종로구  서울특별시 종로구  서울특별시 종로구 종로31가길 25-5   66   
2  LH서울      11  서울특별시    110  종로구  서울특별시 종로구  서울특별시 종로구 종로31가길 25-5   66   
3  LH서울      11  서울특별시    110  종로구  서울특별시 종로구  서울특별시 종로구 종로31가길 25-5   66   
4  LH서울      11  서울특별시    110  종로구  서울특별시 종로구  서울특별시 종로구 종로31가길 25-5   66   

  공급유형명  형명  공급전용면적   공급공용면적 주택유형명  주차수  기본임대보증금  기본월임대료  기본전환보증금  
0  매입임대  19   19.68  12.1124  오피스텔    0   600000  338000        0  
1  매입임대  19   19.69  12.1074  오피스텔    0   600000  312000        0  
2  매입임대  21   21.41  13.2757  오피스텔    0   600000  342000        0  
3  매입임대  21   21.42  13.2706  오피스텔    0   600000  368000        0  
4  매입임대  21   21.94  13.5241  오피스텔    0   600000  363000        0

In [7]:
data.columns=['기관명', '광역시도코드', '지역', '시군구코드', '시군구명', '단지명', '도로명주소', '총세대수',
       '공급유형', '형명', '전용면적', '공급공용면적', '주택유형명', '단지내주차면수', '임대보증금', '임대료',
       '기본전환보증금']

In [8]:
data= data.drop(["형명", "기본전환보증금", "공급공용면적", "광역시도코드"], axis=1)
#data.drop(data.loc[data['단지내주차면수']==0].index, inplace=True)
data.drop(data.loc[data['지역']=='인천광역시'].index, inplace=True)

In [9]:
print(data.shape)

(93668, 13)


# Test: 상가 단지명 특정

In [10]:
test_store= pd.DataFrame({'단지코드':['C1006', 'C2676', 'C2177', 'C2033', 'C1812', 'C2253', 'C1729'],
                     '단지명':['대전둔산1', '모라주공3단지', '읍내동 주공아파트', '달동 주공3단지',
                            '울산화정주공', '명륜주공2차아파트', '춘천효자8단지'],
                     '도로명주소':['대전광역시 서구 월평북로 11',
                              '부산광역시 사상구 모라로192번길 20-21',
                              '충청남도 아산시 외암로 1420',
                              '울산광역시 남구 번영로107번길 5',
                            '울산광역시 동구 월봉10길 16',
                            '강원도 원주시 예술관길 31', '강원도 춘천시 공지로 234-16']})

In [11]:
test_store

단지코드        단지명                     도로명주소
0  C1006      대전둔산1          대전광역시 서구 월평북로 11
1  C2676    모라주공3단지  부산광역시 사상구 모라로192번길 20-21
2  C2177  읍내동 주공아파트         충청남도 아산시 외암로 1420
3  C2033   달동 주공3단지       울산광역시 남구 번영로107번길 5
4  C1812     울산화정주공         울산광역시 동구 월봉10길 16
5  C2253  명륜주공2차아파트           강원도 원주시 예술관길 31
6  C1729    춘천효자8단지        강원도 춘천시 공지로 234-16

# Test: 아파트 단지명 특정

In [12]:
test[test.임대건물구분=='아파트'].isna().sum()

단지코드         0
총세대수         0
임대건물구분       0
지역           0
공급유형         0
전용면적         0
전용면적별세대수     0
공가수          0
자격유형         2
임대보증금        9
임대료          9
지하철역        20
버스정류장        0
단지내주차면수      0
dtype: int64

In [13]:
test_아파트=test[test.임대건물구분=='아파트']

In [14]:
test_아파트[test_아파트.임대료.isnull()]

단지코드  총세대수 임대건물구분     지역  공급유형   전용면적  전용면적별세대수   공가수 자격유형  임대보증금  임대료  \
76    C1006  1505    아파트  대전광역시  영구임대  26.37        10  27.0    D    NaN  NaN   
77    C1006  1505    아파트  대전광역시  영구임대  26.37        10  27.0    D    NaN  NaN   
79    C1006  1505    아파트  대전광역시  영구임대  52.74         6  27.0    D    NaN  NaN   
1005  C2152   120    아파트    강원도  영구임대  24.83        66   9.0    C    NaN  NaN   
1006  C2152   120    아파트    강원도  영구임대  33.84        54   9.0    C    NaN  NaN   
1014  C1267   675    아파트   경상남도  행복주택  16.94        50  38.0    L    NaN  NaN   
1015  C1267   675    아파트   경상남도  행복주택  26.85        66  38.0    L    NaN  NaN   
1016  C1267   675    아파트   경상남도  행복주택  26.85         8  38.0    L    NaN  NaN   
1017  C1267   675    아파트   경상남도  행복주택  36.77       126  38.0    L    NaN  NaN   

      지하철역  버스정류장  단지내주차면수  
76     2.0    5.0    428.0  
77     2.0    5.0    428.0  
79     2.0    5.0    428.0  
1005   0.0    1.0     40.0  
1006   0.0    1.0     40.0  
1014   0.0    1.0    467.0  
1015   0.0    1.0    467.0  
1016   0.0    1.0    467.0  
1017   0.0    1.0    467.0

In [15]:
test_아파트[test_아파트.임대료.isnull()].단지코드.unique()

array(['C1006', 'C2152', 'C1267'], dtype=object)

C1006은 대전둔산1이라는 것을 상가 덕분에 확인한 바 있음 ㅎㅎ 

In [16]:
indata_test=pd.merge(test_아파트, data, on=["지역","단지내주차면수","임대보증금","임대료"], how="inner")

In [17]:
len(set(indata_test.단지코드))

80

In [18]:
indata_test= indata_test[["단지코드", "단지명", "도로명주소"]]

In [19]:
indata_test.단지코드.unique()

array(['C1072', 'C1456', 'C1840', 'C1794', 'C1640', 'C1472', 'C1006',
       'C1083', 'C1311', 'C1603', 'C2411', 'C1604', 'C2535', 'C1040',
       'C2417', 'C1772', 'C1152', 'C2423', 'C1844', 'C1602', 'C1239',
       'C1721', 'C1902', 'C1071', 'C2295', 'C1621', 'C1852', 'C2555',
       'C1297', 'C2323', 'C2215', 'C1830', 'C2608', 'C1215', 'C1158',
       'C1189', 'C1637', 'C1225', 'C2374', 'C2397', 'C1105', 'C1708',
       'C2144', 'C1149', 'C1323', 'C1064', 'C1019', 'C2269', 'C1416',
       'C1250', 'C2675', 'C2436', 'C1253', 'C1166', 'C1135', 'C1216',
       'C1564', 'C2653', 'C1392', 'C2128', 'C1282', 'C1521', 'C2076',
       'C1060', 'C2473', 'C2557', 'C1835', 'C2224', 'C2222', 'C1712',
       'C2267', 'C1474', 'C1633', 'C2113', 'C2175', 'C2268', 'C2551',
       'C2688', 'C2043', 'C2091'], dtype=object)

In [20]:
res = []
for i in indata_test['단지코드'].unique():
    if(indata_test[indata_test['단지코드']==i]['도로명주소'].nunique()!=1): 
        res.append(i)

In [21]:
res

[]

In [22]:
indata_test=indata_test.drop_duplicates(keep="first")

In [23]:
indata_test= indata_test.reset_index(drop=True)

In [24]:
indata_test

단지코드            단지명                   도로명주소
0   C1072  광명역세권 휴먼시아2단지          경기도 광명시 서원로 25
1   C1456         안락휴먼시아  부산광역시 동래구 충렬대로447번길 38
2   C1840      장신휴먼시아3단지   전라북도 익산시 오산면 선화로4길 14
3   C1794      군산미장 휴먼시아        전라북도 군산시 경포천로 34
4   C1640      교동마을 휴먼시아         전라북도 김제시 교정길 36
..    ...            ...                     ...
76  C2268     양산물금LH천년나무     경상남도 양산시 물금읍 청운로 42
77  C2551          호려울마을         세종특별자치시 보람1길 12
78  C2688    함평향교(국민+영구)    전라남도 함평군 대동면 영수길 367
79  C2043     창원석동 LH아파트    경상남도 창원시 진해구 냉천로 125
80  C2091      괴산동부 행복마을     충청북도 괴산군 괴산읍 남산길 43

[81 rows x 3 columns]

In [25]:
test2= test_아파트

In [26]:
d_list=indata_test.단지코드.unique()

In [27]:
for i in d_list:
    test2.drop(test2.loc[test2['단지코드']==i].index, inplace=True)

In [28]:
test2.shape

(379, 14)

In [29]:
indata_test2=pd.merge(test2, data, on=["지역","공급유형","임대보증금","임대료"], how="inner")

In [30]:
indata_test2.head()

단지코드  총세대수_x 임대건물구분   지역  공급유형  전용면적_x  전용면적별세대수  공가수 자격유형       임대보증금  \
0  C1128    1354    아파트  경기도  국민임대   39.79       368  9.0    H  22830000.0   
1  C1128    1354    아파트  경기도  국민임대   39.79       368  9.0    H  22830000.0   
2  C1128    1354    아파트  경기도  국민임대   39.79        30  9.0    H  22830000.0   
3  C1128    1354    아파트  경기도  국민임대   39.79        30  9.0    H  22830000.0   
4  C1128    1354    아파트  경기도  국민임대   39.79        22  9.0    A  22830000.0   

   ...  단지내주차면수_x   기관명  시군구코드  시군구명            단지명            도로명주소 총세대수_y  \
0  ...     1216.0  LH인천    210   광명시  광명역세권 휴먼시아1단지  경기도 광명시 성채안로 26   1354   
1  ...     1216.0  LH인천    210   광명시  광명역세권 휴먼시아2단지   경기도 광명시 서원로 25    754   
2  ...     1216.0  LH인천    210   광명시  광명역세권 휴먼시아1단지  경기도 광명시 성채안로 26   1354   
3  ...     1216.0  LH인천    210   광명시  광명역세권 휴먼시아2단지   경기도 광명시 서원로 25    754   
4  ...     1216.0  LH인천    210   광명시  광명역세권 휴먼시아1단지  경기도 광명시 성채안로 26   1354   

  전용면적_y 주택유형명  단지내주차면수_y  
0  39.79   아파트        999  
1  39.79   아파트        683  
2  39.79   아파트        999  
3  39.79   아파트        683  
4  39.79   아파트        999  

[5 rows x 23 columns]

In [31]:
indata_test2.shape

(581, 23)

In [32]:
indata_test2.단지코드.unique()

array(['C1128', 'C1332', 'C1563', 'C1377', 'C2072', 'C2676', 'C2177',
       'C2033', 'C1812', 'C2253', 'C1294', 'C2646', 'C1729', 'C1855',
       'C2600', 'C2112', 'C2602', 'C2369', 'C1887', 'C1782', 'C1629',
       'C1617', 'C1741', 'C1016', 'C2039', 'C2028', 'C1593', 'C1826',
       'C1827', 'C1495', 'C1457', 'C1525', 'C1623', 'C1147', 'C1692',
       'C1570', 'C2547', 'C2292', 'C1289', 'C1849', 'C1030', 'C1327',
       'C1349', 'C2363', 'C2335', 'C1841', 'C2404', 'C2456', 'C1266',
       'C1267', 'C2189'], dtype=object)

In [33]:
len(set(indata_test2.단지코드))

51

In [34]:
res = []
for i in indata_test2['단지코드'].unique():
    if(indata_test2[indata_test2['단지코드']==i]['도로명주소'].nunique()!=1): 
        res.append(i)

In [35]:
res

['C1128', 'C2676', 'C2177', 'C2253', 'C1887', 'C2028', 'C1327']

In [36]:
for i in res : 
    print('{} 의 주소는 가능성은 {}개 {}'.format(i, indata_test2[indata_test2['단지코드']==i]['도로명주소'].nunique(),indata_test2[indata_test2['단지코드']==i]['도로명주소'].unique()))

C1128 의 주소는 가능성은 2개 ['경기도 광명시 성채안로 26' '경기도 광명시 서원로 25']
C2676 의 주소는 가능성은 7개 ['부산광역시 영도구 상리로 1' '부산광역시 영도구 동삼북로 2' '부산광역시 북구 효열로 76' '부산광역시 북구 덕천로 74'
 '부산광역시 해운대구 신반송로 200' '부산광역시 사상구 모라로192번길 20-21' '부산광역시 사상구 모라로110번길 121']
C2177 의 주소는 가능성은 3개 ['충청남도 천안시 서북구 쌍용11길 57' '충청남도 공주시 우금티로 709-12' '충청남도 아산시 외암로 1420']
C2253 의 주소는 가능성은 2개 ['강원도 원주시 예술관길 31' '강원도 속초시 먹거리길 17']
C1887 의 주소는 가능성은 2개 ['대구광역시 달성군 다사읍 매곡로14길 11' '대구광역시 달성군 다사읍 매곡로14길 13']
C2028 의 주소는 가능성은 3개 ['경상남도 김해시 율하2로 164' '경상남도 김해시 율하2로 178' '경상남도 김해시 율하1로 7']
C1327 의 주소는 가능성은 2개 ['대전광역시 중구 목중로39번길 35-14' '대전광역시 서구 계백로 1111']


<상가에서 찾은 아이들>
- C2676: 모라주공3단지/ 부산광역시 사상구 모라로192번길 20-21
- C2177: 읍내동 주공아파트 / 충청남도 아산시 외암로 1420
- C2253: 명륜주공2차아파트/ 강원도 원주시 예술관길 31

어차피 나중에 pd.concat할거니까 드랍하자 ㅎㅎ

In [37]:
indata_test2= indata_test2[indata_test2.단지코드!="C2676"]
indata_test2= indata_test2[indata_test2.단지코드!="C2177"]
indata_test2= indata_test2[indata_test2.단지코드!="C2253"]

In [38]:
indata_test2[indata_test2.단지코드=="C1128"].head()

단지코드  총세대수_x 임대건물구분   지역  공급유형  전용면적_x  전용면적별세대수  공가수 자격유형       임대보증금  \
0  C1128    1354    아파트  경기도  국민임대   39.79       368  9.0    H  22830000.0   
1  C1128    1354    아파트  경기도  국민임대   39.79       368  9.0    H  22830000.0   
2  C1128    1354    아파트  경기도  국민임대   39.79        30  9.0    H  22830000.0   
3  C1128    1354    아파트  경기도  국민임대   39.79        30  9.0    H  22830000.0   
4  C1128    1354    아파트  경기도  국민임대   39.79        22  9.0    A  22830000.0   

   ...  단지내주차면수_x   기관명  시군구코드  시군구명            단지명            도로명주소 총세대수_y  \
0  ...     1216.0  LH인천    210   광명시  광명역세권 휴먼시아1단지  경기도 광명시 성채안로 26   1354   
1  ...     1216.0  LH인천    210   광명시  광명역세권 휴먼시아2단지   경기도 광명시 서원로 25    754   
2  ...     1216.0  LH인천    210   광명시  광명역세권 휴먼시아1단지  경기도 광명시 성채안로 26   1354   
3  ...     1216.0  LH인천    210   광명시  광명역세권 휴먼시아2단지   경기도 광명시 서원로 25    754   
4  ...     1216.0  LH인천    210   광명시  광명역세권 휴먼시아1단지  경기도 광명시 성채안로 26   1354   

  전용면적_y 주택유형명  단지내주차면수_y  
0  39.79   아파트        999  
1  39.79   아파트        683  
2  39.79   아파트        999  
3  39.79   아파트        683  
4  39.79   아파트        999  

[5 rows x 23 columns]

- C1128: 광명역세권 휴먼시아1단지

In [39]:
indata_test2=indata_test2[indata_test2.단지코드!="C1128"]

In [40]:
indata_test2[indata_test2.단지코드=="C1887"]

단지코드  총세대수_x 임대건물구분     지역  공급유형  전용면적_x  전용면적별세대수   공가수 자격유형  \
209  C1887     801    아파트  대구광역시  국민임대   36.19        60  21.0    A   
210  C1887     801    아파트  대구광역시  국민임대   36.19        60  21.0    A   
211  C1887     801    아파트  대구광역시  국민임대   36.19        60  21.0    A   
212  C1887     801    아파트  대구광역시  국민임대   36.19        60  21.0    A   
213  C1887     801    아파트  대구광역시  국민임대   36.23        56  21.0    A   
214  C1887     801    아파트  대구광역시  국민임대   36.23        56  21.0    A   
215  C1887     801    아파트  대구광역시  국민임대   36.23        56  21.0    A   
216  C1887     801    아파트  대구광역시  국민임대   36.23        56  21.0    A   
217  C1887     801    아파트  대구광역시  국민임대   36.30        30  21.0    A   
218  C1887     801    아파트  대구광역시  국민임대   36.30        30  21.0    A   
219  C1887     801    아파트  대구광역시  국민임대   36.30        30  21.0    A   
220  C1887     801    아파트  대구광역시  국민임대   36.30        30  21.0    A   
221  C1887     801    아파트  대구광역시  국민임대   36.63       120  21.0    A   
222  C1887     801    아파트  대구광역시  국민임대   36.63       120  21.0    A   
223  C1887     801    아파트  대구광역시  국민임대   36.63       120  21.0    A   
224  C1887     801    아파트  대구광역시  국민임대   36.63       120  21.0    A   
225  C1887     801    아파트  대구광역시  국민임대   36.63        15  21.0    A   
226  C1887     801    아파트  대구광역시  국민임대   36.63        15  21.0    A   
227  C1887     801    아파트  대구광역시  국민임대   36.63        15  21.0    A   
228  C1887     801    아파트  대구광역시  국민임대   36.63        15  21.0    A   
229  C1887     801    아파트  대구광역시  국민임대   46.19       171  21.0    A   
230  C1887     801    아파트  대구광역시  국민임대   46.19        41  21.0    A   
231  C1887     801    아파트  대구광역시  국민임대   51.77        84  21.0    A   
232  C1887     801    아파트  대구광역시  국민임대   51.77        86  21.0    A   
233  C1887     801    아파트  대구광역시  국민임대   51.77        42  21.0    A   
234  C1887     801    아파트  대구광역시  국민임대   59.29        48  21.0    A   
235  C1887     801    아파트  대구광역시  국민임대   59.29        48  21.0    A   
236  C1887     801    아파트  대구광역시  국민임대   59.57        48  21.0    A   
237  C1887     801    아파트  대구광역시  국민임대   59.57        48  21.0    A   

          임대보증금  ...  단지내주차면수_x     기관명  시군구코드  시군구명        단지명  \
209  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
210  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
211  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
212  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
213  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
214  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
215  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
216  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
217  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
218  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
219  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
220  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
221  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
222  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
223  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
224  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
225  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
226  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
227  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
228  19472000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
229  25964000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
230  25964000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
231  28559000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
232  28559000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
233  28559000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
234  32454000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   
235  32454000.0  ...      611.0  LH대구경북    710   달성군  대구다사주공아파트   

- C1887: 대구다사주공아파트이고 둘다 맞는데;; 심지어 저거 주소 두개의 총세대수를 합쳐서 801임. 
- 전용면적별로 다르게 넣어주면 될것 같음... 일단 살리겠음...

In [41]:
indata_test2[indata_test2.단지코드=="C2028"].head()

단지코드  총세대수_x 임대건물구분    지역  공급유형  전용면적_x  전용면적별세대수   공가수 자격유형  \
276  C2028    1267    아파트  경상남도  국민임대   51.93       119  26.0    A   
277  C2028    1267    아파트  경상남도  국민임대   51.93       119  26.0    A   
278  C2028    1267    아파트  경상남도  국민임대   51.93       119  26.0    A   
279  C2028    1267    아파트  경상남도  국민임대   51.93        59  26.0    A   
280  C2028    1267    아파트  경상남도  국민임대   51.93        59  26.0    A   

          임대보증금  ...  단지내주차면수_x   기관명  시군구코드  시군구명               단지명  \
276  24969000.0  ...     1280.0  LH경남    250   김해시   율현마을 주공 13단지아파트   
277  24969000.0  ...     1280.0  LH경남    250   김해시  율현마을 주공 12단지 아파트   
278  24969000.0  ...     1280.0  LH경남    250   김해시    율곡마을 주공2단지 아파트   
279  24969000.0  ...     1280.0  LH경남    250   김해시   율현마을 주공 13단지아파트   
280  24969000.0  ...     1280.0  LH경남    250   김해시  율현마을 주공 12단지 아파트   

                 도로명주소 총세대수_y 전용면적_y 주택유형명  단지내주차면수_y  
276  경상남도 김해시 율하2로 164   1267  51.93   아파트       1282  
277  경상남도 김해시 율하2로 178    757  51.93   아파트        761  
278    경상남도 김해시 율하1로 7    944  51.78   아파트        958  
279  경상남도 김해시 율하2로 164   1267  51.93   아파트       1282  
280  경상남도 김해시 율하2로 178    757  51.93   아파트        761  

[5 rows x 23 columns]

- C2028: 율현마을 주공 13단지아파트 / 경상남도 김해시 율하2로 164

In [42]:
indata_test2=indata_test2[indata_test2.단지코드!="C2028"]

In [43]:
test[test.단지코드=="C1327"]

단지코드  총세대수 임대건물구분     지역  공급유형   전용면적  전용면적별세대수   공가수 자격유형       임대보증금  \
891  C1327    98    아파트  대전광역시  행복주택  16.83        30  13.0    J  18800000.0   
892  C1327    98    아파트  대전광역시  행복주택  26.51        10  13.0    J  28400000.0   
893  C1327    98    아파트  대전광역시  행복주택  26.51        10  13.0    J  28400000.0   
894  C1327    98    아파트  대전광역시  행복주택  36.92        48  13.0    J  36800000.0   

          임대료  지하철역  버스정류장  단지내주차면수  
891   94000.0   NaN   17.0     98.0  
892  142000.0   NaN   17.0     98.0  
893  142000.0   NaN   17.0     98.0  
894  184000.0   NaN   17.0     98.0

In [44]:
data[data.단지명=="대전목동2 행복주택"]

기관명     지역  시군구코드 시군구명         단지명                   도로명주소  총세대수  \
55263  LH대전충남  대전광역시    140   중구  대전목동2 행복주택  대전광역시 중구 목중로39번길 35-14    98   
55264  LH대전충남  대전광역시    140   중구  대전목동2 행복주택  대전광역시 중구 목중로39번길 35-14    98   
55265  LH대전충남  대전광역시    140   중구  대전목동2 행복주택  대전광역시 중구 목중로39번길 35-14    98   

       공급유형   전용면적 주택유형명  단지내주차면수     임대보증금     임대료  
55263  행복주택  16.83   아파트        0  18800000   94000  
55264  행복주택  26.51   아파트        0  28400000  142000  
55265  행복주택  36.92   아파트        0  36800000  184000

- C1327: 대전목동2 행복주택	/ 대전광역시 중구 목중로39번길 35-14

In [45]:
indata_test2=indata_test2[indata_test2.단지코드!="C1327"]

In [46]:
중복1=pd.DataFrame({'단지코드':['C1128', 'C2028',
                                'C2676', 'C2177', 'C2253', 'C1327'],
                     '단지명':['광명역세권 휴먼시아1단지', '율현마을 주공 13단지아파트', 
                               '모라주공3단지', 
                               '읍내동 주공아파트',
                               '명륜주공2차아파트', '대전목동2 행복주택'],
                     '도로명주소':['경기도 광명시 성채안로 26','경상남도 김해시 율하2로 164',
                                   '부산광역시 사상구 모라로192번길 20-21', 
                                '충청남도 아산시 외암로 1420', 
                                '강원도 원주시 예술관길 31', '대전광역시 중구 목중로39번길 35-14']})

In [47]:
중복1

단지코드              단지명                     도로명주소
0  C1128    광명역세권 휴먼시아1단지           경기도 광명시 성채안로 26
1  C2028  율현마을 주공 13단지아파트         경상남도 김해시 율하2로 164
2  C2676          모라주공3단지  부산광역시 사상구 모라로192번길 20-21
3  C2177        읍내동 주공아파트         충청남도 아산시 외암로 1420
4  C2253        명륜주공2차아파트           강원도 원주시 예술관길 31
5  C1327       대전목동2 행복주택    대전광역시 중구 목중로39번길 35-14

In [48]:
indata_test2= indata_test2[["단지코드", "단지명", "도로명주소"]]

In [49]:
indata_test2=indata_test2.drop_duplicates(keep='first')

In [50]:
indata_test2= indata_test2.reset_index(drop=True)

In [51]:
indata_test2=pd.concat([indata_test2, 중복1], axis=0).reset_index(drop=True)

In [52]:
indata_test2

단지코드                단지명                      도로명주소
0   C1332           봇들마을 6단지       경기도 성남시 분당구 동판교로 212
1   C1563           백현마을 4단지        경기도 성남시 분당구 동판교로 91
2   C1377             거제일운주공     경상남도 거제시 일운면 거제대로 2591
3   C2072          남해평리 휴먼시아       경상남도 남해군 남해읍 스포츠로 20
4   C2033           달동 주공3단지        울산광역시 남구 번영로107번길 5
5   C1812             울산화정주공          울산광역시 동구 월봉10길 16
6   C1294         원주무실 주공1단지             강원도 원주시 시청로 68
7   C2646         원주태장 주공4단지         강원도 원주시 흥양로51번길 20
8   C1729           춘천효자8 단지         강원도 춘천시 공지로 234-16
9   C1855           음성삼성휴먼시아      충청북도 음성군 삼성면 덕정로 29-7
10  C2600       익산함열 휴먼시아1단지       전라북도 익산시 함열읍 함열5길 28
11  C2112          구평휴먼시아아파트          경상북도 구미시 인동54길 25
12  C2602       김제검산 휴먼시아3단지           전라북도 김제시 금성로 140
13  C2369         논산취암 주공2단지       충청남도 논산시 논산대로348번길 7
14  C1887          대구다사주공아파트    대구광역시 달성군 다사읍 매곡로14길 11
15  C1887          대구다사주공아파트    대구광역시 달성군 다사읍 매곡로14길 13
16  C1782         삼척건지 주공아파트           강원도 삼척시 엑스포로 174
17  C1629          삼척도계주공아파트        강원도 삼척시 도계읍 도계로 134
18  C1617      양산신도시주공3단지아파트           경상남도 양산시 양주로 153
19  C1741          성현마을주공아파트    경기도 용인시 수지구 죽전로193번길 11
20  C1016               도담마을     경기도 용인시 수지구 죽전로27번길 60
21  C2039        율하휴먼시아 10단지         대구광역시 동구 율하동로8길 16
22  C1593        제천장락 4주공아파트          충청북도 제천시 내토로59길 7
23  C1826         진천덕산 주공아파트       충청북도 진천군 덕산읍 용몽1길 11
24  C1827          진천신정주공아파트       충청북도 진천군 진천읍 문화2길 14
25  C1495         퇴계주공7단지아파트        강원도 춘천시 승지골길16번길 14
26  C1457           울산 우정1단지         울산광역시 중구 종가로 668-9
27  C1525        미리내마을 4-4단지           경기도 남양주시 별내3로 23
28  C1623            군위서부1단지       경상북도 군위군 군위읍 동산새길 50
29  C1147          창원현동LH아파트      경상남도 창원시 마산합포구 현동3길 1
30  C1692  대구테크노폴리스 천년나무 1단지   대구광역시 달성군 현풍읍 테크노북로2길 10
31  C1570         충북혁신도시 1단지        충청북도 음성군 맹동면 이수로 31
32  C2547        별사랑마을 2-5단지          경기도 남양주시 별내5로 189
33  C2292           새뜸마을 8단지           세종특별자치시 새롬중앙로 20
34  C1289          춘천우두 국민임대            강원도 춘천시 우두로 130
35  C1849            미디어촌5단지            강원도 강릉시 선수촌로 11
36  C1030          용인김량장(행복)   경기도 용인시 처인구 금령로40번길 15-5
37  C1349          LH송화마을8단지            광주광역시 남구 노대실로 9
38  C2363   부산정관 행복주택(A-4BL)      부산광역시 기장군 정관읍 정관1로 51
39  C2335     익산부송1 증축 주거복지동         전라북도 익산시 무왕로22길 12
40  C1841            광주첨단H-3  광주광역시 광산구 임방울대로800번길 11-5
41  C2404          여수관문 행복주택           전라남도 여수시 관문1길 14
42  C2456            서귀포성산LH  제주특별자치도 서귀포시 성산읍 고성동서로 33
43  C1266   충주호암A4-1블록(국민임대)         충청북도 충주시 호암수청2로 56
44  C1267            창원가포1단지   경상남도 창원시 마산합포구 가포본동5길 37
45  C2189     정읍첨단1 주거행복복지센터            전라북도 정읍시 첨단3로 7
46  C1128      광명역세권 휴먼시아1단지            경기도 광명시 성채안로 26
47  C2028    율현마을 주공 13단지아파트          경상남도 김해시 율하2로 164
48  C2676            모라주공3단지   부산광역시 사상구 모라로192번길 20-21
49  C2177          읍내동 주공아파트          충청남도 아산시 외암로 1420
50  C2253          명륜주공2차아파트            강원도 원주시 예술관길 31
51  C1327         대전목동2 행복주택     대전광역시 중구 목중로39번길 35-14

In [53]:
d_list=np.append(d_list, indata_test2.단지코드.unique())

In [54]:
len(d_list)

131

In [55]:
test3=test2

In [56]:
test3

단지코드  총세대수 임대건물구분    지역  공급유형   전용면적  전용면적별세대수   공가수 자격유형       임대보증금  \
8     C1128  1354    아파트   경기도  국민임대  39.79       368   9.0    H  22830000.0   
9     C1128  1354    아파트   경기도  국민임대  39.79        30   9.0    H  22830000.0   
10    C1128  1354    아파트   경기도  국민임대  39.79        22   9.0    A  22830000.0   
11    C1128  1354    아파트   경기도  국민임대  46.81        60   9.0    A  36048000.0   
12    C1128  1354    아파트   경기도  국민임대  46.90       116   9.0    A  36048000.0   
...     ...   ...    ...   ...   ...    ...       ...   ...  ...         ...   
1017  C1267   675    아파트  경상남도  행복주택  36.77       126  38.0    L         NaN   
1018  C2189   382    아파트  전라북도  국민임대  29.19        96  45.0    H   6872000.0   
1019  C2189   382    아파트  전라북도  국민임대  29.19        20  45.0    H   6872000.0   
1020  C2189   382    아파트  전라북도  국민임대  39.45       202  45.0    H  13410000.0   
1021  C2189   382    아파트  전라북도  국민임대  46.23        60  45.0    H  18689000.0   

           임대료  지하철역  버스정류장  단지내주차면수  
8     189840.0   0.0    3.0   1216.0  
9     189840.0   0.0    3.0   1216.0  
10    189840.0   0.0    3.0   1216.0  
11    249930.0   0.0    3.0   1216.0  
12    249930.0   0.0    3.0   1216.0  
...        ...   ...    ...      ...  
1017       NaN   0.0    1.0    467.0  
1018  106400.0   0.0    2.0    300.0  
1019  106400.0   0.0    2.0    300.0  
1020  144600.0   0.0    2.0    300.0  
1021  166500.0   0.0    2.0    300.0  

[379 rows x 14 columns]

In [57]:
for i in d_list:
    test3.drop(test3.loc[test3['단지코드']==i].index, inplace=True)

In [58]:
indata_test3=pd.merge(test3, data, on=["지역","총세대수","임대보증금","임대료"], how="inner")

In [59]:
indata_test3

단지코드  총세대수 임대건물구분     지역     공급유형_x  전용면적_x  전용면적별세대수   공가수 자격유형  \
0   C2318   420    아파트  부산광역시  공공임대(50년)   49.94       276   1.0    A   
1   C2318   420    아파트  부산광역시  공공임대(50년)   49.94       276   1.0    A   
2   C2318   420    아파트  부산광역시  공공임대(50년)   49.94       144   1.0    A   
3   C2318   420    아파트  부산광역시  공공임대(50년)   49.94       144   1.0    A   
4   C1318   312    아파트   충청남도  공공임대(50년)   39.51       135   1.0    A   
5   C1318   312    아파트   충청남도  공공임대(50년)   39.72        18   1.0    A   
6   C1318   312    아파트   충청남도  공공임대(50년)   49.99       159   1.0    A   
7   C2430   687    아파트  대전광역시   공공임대(분납)   51.88       238  13.0    A   
8   C2430   687    아파트  대전광역시   공공임대(분납)   59.69       137  13.0    A   
9   C2430   687    아파트  대전광역시   공공임대(분납)   59.99        37  13.0    A   
10  C1494   728    아파트    강원도  공공임대(10년)   74.95        66  10.0    A   
11  C1494   728    아파트    강원도  공공임대(10년)   84.91       582  10.0    A   
12  C1494   728    아파트    강원도  공공임대(10년)   84.91       582  10.0    A   
13  C1494   728    아파트    강원도  공공임대(10년)   84.94        80  10.0    A   
14  C1494   728    아파트    강원도  공공임대(10년)   84.94        80  10.0    A   
15  C2131   538    아파트   충청남도  공공임대(10년)   51.88        36   5.0    A   
16  C2131   538    아파트   충청남도   공공임대(분납)   51.84        77   5.0    A   
17  C2131   538    아파트   충청남도   공공임대(분납)   59.44       106   5.0    A   
18  C2691   765    아파트    경기도  공공임대(10년)   74.89        25  25.0    A   
19  C2691   765    아파트    경기도  공공임대(10년)   74.89        25  25.0    A   
20  C2691   765    아파트    경기도  공공임대(10년)   74.95       150  25.0    A   
21  C2691   765    아파트    경기도  공공임대(10년)   74.95       150  25.0    A   
22  C2691   765    아파트    경기도  공공임대(10년)   84.84       590  25.0    A   
23  C2303   963    아파트    경기도  공공임대(10년)   74.93        89  16.0    A   
24  C2303   963    아파트    경기도  공공임대(10년)   74.93        89  16.0    A   
25  C2303   963    아파트    경기도  공공임대(10년)   84.76       165  16.0    A   
26  C2303   963    아파트    경기도  공공임대(10년)   84.91       400  16.0    A   
27  C2303   963    아파트    경기도  공공임대(10년)   84.91       150  16.0    A   
28  C2303   963    아파트    경기도  공공임대(10년)   84.91        70  16.0    A   
29  C1496   565    아파트   충청북도  공공임대(10년)   75.90       118  11.0    A   
30  C1496   565    아파트   충청북도  공공임대(10년)   84.64        98  11.0    A   
31  C1496   565    아파트   충청북도  공공임대(10년)   84.64        98  11.0    A   
32  C1496   565    아파트   충청북도  공공임대(10년)   84.88       349  11.0    A   
33  C1496   565    아파트   충청북도  공공임대(10년)   84.88       349  11.0    A   
34  C2451   395    아파트  대구광역시  공공임대(10년)   59.18        23   2.0    A   
35  C2451   395    아파트  대구광역시  공공임대(10년)   59.18        23   2.0    A   
36  C2451   395    아파트  대구광역시  공공임대(10년)   59.18        23   2.0    A   
37  C2451   395    아파트  대구광역시  공공임대(10년)   59.18        23   2.0    A   
38  C2451   395    아파트  대구광역시  공공임대(10년)   59.19       350   2.0    A   
39  C2451   395    아파트  대구광역시  공공임대(10년)   59.19       350   2.0    A   
40  C2451   395    아파트  대구광역시  공공임대(10년)   59.19       350   2.0    A   
41  C2451   395    아파트  대구광역시  공공임대(10년)   59.19       350   2.0    A   
42  C2451   395    아파트  대구광역시  공공임대(10년)   59.51         9   2.0    A   
43  C2451   395    아파트  대구광역시  공공임대(10년)   59.51         9   2.0    A   
44  C2451   395    아파트  대구광역시  공공임대(10년)   59.51         9   2.0    A   
45  C2451   395    아파트  대구광역시  공공임대(10년)   59.51         9   2.0    A   
46  C2451   395    아파트  대구광역시  공공임대(10년)   59.92        13   2.0    A   
47  C2451   395    아파트  대구광역시  공공임대(10년)   59.92        13   2.0    A   
48  C2451   395    아파트  대구광역시  공공임대(10년)   59.92        13   2.0    A   
49  C2451   395    아파트  대구광역시  공공임대(10년)   59.92        13   2.0    A   

         임대보증금  ...  단지내주차면수_x     기관명  시군구코드     시군구명                단지명  \
0   21801000.0  ...      210.0  LH부산울산    350     해운대구     부산해운대주공아파트 1단지   
1   21801000.0  ...      210.0  LH부산울산    350     해운대구     부산해운대주공아파트 1단지   
2   21801000.0  ...      210.0  LH부산울산    350     

In [60]:
indata_test3.단지코드.unique()

array(['C2318', 'C1318', 'C2430', 'C1494', 'C2131', 'C2691', 'C2303',
       'C1496', 'C2451'], dtype=object)

In [61]:
len(indata_test3.단지코드.unique())

9

In [62]:
res = []
for i in indata_test3['단지코드'].unique():
    if(indata_test3[indata_test3['단지코드']==i]['도로명주소'].nunique()!=1): 
        res.append(i)

In [63]:
res

[]

In [64]:
indata_test3= indata_test3[["단지코드", "단지명", "도로명주소"]]

In [65]:
indata_test3= indata_test3.drop_duplicates(keep='first').reset_index(drop=True)

In [66]:
indata_test3

단지코드                단지명                  도로명주소
0  C2318     부산해운대주공아파트 1단지   부산광역시 해운대구 대천로67번길 4
1  C1318           쌍용마을 뜨란채  충청남도 천안시 서북구 봉서산샛길 64
2  C2430  대전노은3 A3블록 공공임대주택  대전광역시 유성구 지족로148번길 22
3  C1494   푸른숨 3단지 10년 공공임대         강원도 원주시 배울로 50
4  C2131   논산내동2(05,국민) 1단지   충청남도 논산시 시민로132번길 10
5  C2691        평택소사벌 B-5BL        경기도 평택시 평남로 962
6  C2303     (NHF제3호)안성아양B4         경기도 안성시 아양로 73
7  C1496         제천강저 B-2블록     충청북도 제천시 청풍호로7길 76
8  C2451        대구대곡수목원풍경그린      대구광역시 달서구 갈밭남로 64

In [67]:
test4=test3

In [68]:
test4

단지코드  총세대수 임대건물구분     지역       공급유형   전용면적  전용면적별세대수   공가수 자격유형  \
159   C2318   420    아파트  부산광역시  공공임대(50년)  49.94       276   1.0    A   
160   C2318   420    아파트  부산광역시  공공임대(50년)  49.94       144   1.0    A   
165   C2523   371    아파트    강원도       국민임대  36.65        98   2.0    A   
166   C2523   371    아파트    강원도       국민임대  46.90       208   2.0    A   
167   C2523   371    아파트    강원도       국민임대  51.93        65   2.0    A   
...     ...   ...    ...    ...        ...    ...       ...   ...  ...   
993   C1038   554    아파트   충청북도       행복주택  26.76        12  34.0    K   
994   C1038   554    아파트   충청북도       행복주택  36.76       208  34.0    K   
995   C1038   554    아파트   충청북도       행복주택  36.95        18  34.0    K   
1005  C2152   120    아파트    강원도       영구임대  24.83        66   9.0    C   
1006  C2152   120    아파트    강원도       영구임대  33.84        54   9.0    C   

           임대보증금       임대료  지하철역  버스정류장  단지내주차면수  
159   21801000.0  234350.0   2.0    8.0    210.0  
160   21801000.0  234350.0   2.0    8.0    210.0  
165   14943000.0  126030.0   0.0   13.0    281.0  
166   21442000.0  178800.0   0.0   13.0    281.0  
167   25992000.0  215710.0   0.0   13.0    281.0  
...          ...       ...   ...    ...      ...  
993   57060000.0   29710.0   0.0    1.0    387.0  
994   79833000.0   41580.0   0.0    1.0    387.0  
995   79833000.0   41580.0   0.0    1.0    387.0  
1005         NaN       NaN   0.0    1.0     40.0  
1006         NaN       NaN   0.0    1.0     40.0  

[96 rows x 14 columns]

In [69]:
d_list=np.append(d_list, indata_test3.단지코드.unique())
len(d_list)

140

In [70]:
for i in d_list:
    test4.drop(test4.loc[test4['단지코드']==i].index, inplace=True)

In [71]:
len(test4.단지코드.unique())

10

In [72]:
indata_test4=pd.merge(test4, data, on=["지역","공급유형","총세대수"], how="inner")

In [73]:
indata_test4.shape

(69, 24)

In [74]:
indata_test4.단지코드.unique()

array(['C2523', 'C1003', 'C2152'], dtype=object)

In [75]:
res = []
for i in indata_test4['단지코드'].unique():
    if(indata_test4[indata_test4['단지코드']==i]['도로명주소'].nunique()!=1): 
        res.append(i)

In [76]:
res

[]

In [77]:
indata_test4= indata_test4[["단지코드", "단지명", "도로명주소"]]
indata_test4= indata_test4.drop_duplicates(keep='first').reset_index(drop=True)

In [78]:
indata_test4

단지코드              단지명                     도로명주소
0  C2523        속초청대주공아파트           강원도 속초시 소평로 112
1  C1003  김해진영 LH천년나무 2단지  경상남도 김해시 진영읍 김해대로 464-48
2  C2152  화천신읍(공공실버) 영구임대      강원도 화천군 화천읍 산천어길 196

In [79]:
d_list=np.append(d_list, indata_test4.단지코드.unique())
len(d_list)

143

In [80]:
test5=test4

In [81]:
for i in d_list:
    test5.drop(test5.loc[test5['단지코드']==i].index, inplace=True)

In [82]:
len(test5.단지코드.unique())

7

In [83]:
indata_test5=pd.merge(test5, data, on=["지역","공급유형",'단지내주차면수'], how="inner")

In [84]:
indata_test5.단지코드.unique()

array(['C1229', 'C2174'], dtype=object)

In [85]:
res = []
for i in indata_test5['단지코드'].unique():
    if(indata_test5[indata_test5['단지코드']==i]['도로명주소'].nunique()!=1): 
        res.append(i)

In [86]:
res

[]

In [87]:
indata_test5= indata_test5[["단지코드", "단지명", "도로명주소"]]
indata_test5= indata_test5.drop_duplicates(keep='first').reset_index(drop=True)

In [88]:
indata_test5

단지코드        단지명                  도로명주소
0  C1229  세종서창 행복주택  세종특별자치시 조치원읍 조치원중고길 7
1  C2174      울산송정1      울산광역시 북구 박상진4로 56

In [89]:
d_list=np.append(d_list, indata_test5.단지코드.unique())
len(d_list)

145

In [90]:
test6=test5

In [91]:
for i in d_list:
    test6.drop(test6.loc[test6['단지코드']==i].index, inplace=True)

In [92]:
len(test6.단지코드.unique())

5

In [93]:
test6.단지코드.unique()

array(['C2497', 'C1749', 'C1414', 'C1683', 'C1038'], dtype=object)

In [94]:
pd.merge(test_아파트[test_아파트.단지코드=="C2497"], data, on=["지역", "전용면적","임대보증금", "임대료"], how='inner')

단지코드  총세대수_x 임대건물구분    지역     공급유형_x   전용면적  전용면적별세대수   공가수 자격유형  \
0  C2497     488    아파트  경상북도  공공임대(10년)  59.95        89  37.0    A   
1  C2497     488    아파트  경상북도  공공임대(10년)  59.95        89  37.0    A   
2  C2497     488    아파트  경상북도  공공임대(10년)  59.99        41  37.0    A   
3  C2497     488    아파트  경상북도  공공임대(10년)  59.99        41  37.0    A   
4  C2497     488    아파트  경상북도  공공임대(10년)  74.90       215  37.0    A   
5  C2497     488    아파트  경상북도  공공임대(10년)  74.90       215  37.0    A   
6  C2497     488    아파트  경상북도  공공임대(10년)  74.90       215  37.0    A   
7  C2497     488    아파트  경상북도  공공임대(10년)  74.98        21  37.0    A   

        임대보증금  ...  단지내주차면수_x     기관명  시군구코드  시군구명                      단지명  \
0  31000000.0  ...      624.0  LH대구경북    150   김천시             경북혁신도시 Ab9블록   
1  31000000.0  ...      624.0  LH대구경북    150   김천시  경북혁신 Ab-8블록 보금자리주택 공공임대   
2  31000000.0  ...      624.0  LH대구경북    150   김천시             경북혁신도시 Ab9블록   
3  31000000.0  ...      624.0  LH대구경북    150   김천시  경북혁신 Ab-8블록 보금자리주택 공공임대   
4  35800000.0  ...      624.0  LH대구경북    150   김천시             경북혁신도시 Ab9블록   
5  35800000.0  ...      624.0  LH대구경북    150   김천시             경북혁신도시 Ab9블록   
6  35800000.0  ...      624.0  LH대구경북    150   김천시  경북혁신 Ab-8블록 보금자리주택 공공임대   
7  35800000.0  ...      624.0  LH대구경북    150   김천시             경북혁신도시 Ab9블록   

              도로명주소 총세대수_y 공급유형_y 주택유형명  단지내주차면수_y  
0  경상북도 김천시 용전3로 10    292  10년임대   아파트          0  
1  경상북도 김천시 혁신4로 72    301  10년임대   아파트          0  
2  경상북도 김천시 용전3로 10    292  10년임대   아파트          0  
3  경상북도 김천시 혁신4로 72    301  10년임대   아파트          0  
4  경상북도 김천시 용전3로 10      1  10년임대   아파트          0  
5  경상북도 김천시 용전3로 10    292  10년임대   아파트          0  
6  경상북도 김천시 혁신4로 72    301  10년임대   아파트          0  
7  경상북도 김천시 용전3로 10    292  10년임대   아파트          0  

[8 rows x 23 columns]

- C2497: 경북혁신도시 Ab9블록 / 경상북도 김천시 용전3로 10

In [95]:
pd.merge(test_아파트[test_아파트.단지코드=="C1749"], data, on=["지역", "전용면적"], how='inner')

단지코드  총세대수_x 임대건물구분   지역 공급유형_x   전용면적  전용면적별세대수   공가수 자격유형     임대보증금_x  \
0   C1749     848    아파트  경기도   행복주택  20.59       200  34.0    J  21999000.0   
1   C1749     848    아파트  경기도   행복주택  20.59       200  34.0    J  21999000.0   
2   C1749     848    아파트  경기도   행복주택  20.59        24  34.0    J  21999000.0   
3   C1749     848    아파트  경기도   행복주택  20.59        24  34.0    J  21999000.0   
4   C1749     848    아파트  경기도   행복주택  22.59        72  34.0    J  24137000.0   
5   C1749     848    아파트  경기도   행복주택  22.59        72  34.0    J  24137000.0   
6   C1749     848    아파트  경기도   행복주택  22.59        72  34.0    J  24137000.0   
7   C1749     848    아파트  경기도   행복주택  26.44       167  34.0    J  26933000.0   
8   C1749     848    아파트  경기도   행복주택  26.44       167  34.0    J  26933000.0   
9   C1749     848    아파트  경기도   행복주택  26.44       167  34.0    J  26933000.0   
10  C1749     848    아파트  경기도   행복주택  26.44        15  34.0    J  26933000.0   
11  C1749     848    아파트  경기도   행복주택  26.44        15  34.0    J  26933000.0   
12  C1749     848    아파트  경기도   행복주택  26.44        15  34.0    J  26933000.0   
13  C1749     848    아파트  경기도   행복주택  36.42       129  34.0    J  37090000.0   
14  C1749     848    아파트  경기도   행복주택  36.42       129  34.0    J  37090000.0   
15  C1749     848    아파트  경기도   행복주택  36.42       129  34.0    J  37090000.0   
16  C1749     848    아파트  경기도   행복주택  36.42       129  34.0    J  37090000.0   
17  C1749     848    아파트  경기도   행복주택  36.42       129  34.0    J  37090000.0   
18  C1749     848    아파트  경기도   행복주택  36.42       129  34.0    J  37090000.0   
19  C1749     848    아파트  경기도   행복주택  36.42       129  34.0    J  37090000.0   
20  C1749     848    아파트  경기도   행복주택  36.42         9  34.0    J  37090000.0   
21  C1749     848    아파트  경기도   행복주택  36.42         9  34.0    J  37090000.0   
22  C1749     848    아파트  경기도   행복주택  36.42         9  34.0    J  37090000.0   
23  C1749     848    아파트  경기도   행복주택  36.42         9  34.0    J  37090000.0   
24  C1749     848    아파트  경기도   행복주택  36.42         9  34.0    J  37090000.0   
25  C1749     848    아파트  경기도   행복주택  36.42         9  34.0    J  37090000.0   
26  C1749     848    아파트  경기도   행복주택  36.42         9  34.0    J  37090000.0   
27  C1749     848    아파트  경기도   행복주택  36.70       200  34.0    J  37378000.0   
28  C1749     848    아파트  경기도   행복주택  36.70       200  34.0    J  37378000.0   
29  C1749     848    아파트  경기도   행복주택  36.70       200  34.0    J  37378000.0   
30  C1749     848    아파트  경기도   행복주택  36.70       200  34.0    J  37378000.0   
31  C1749     848    아파트  경기도   행복주택  36.70       200  34.0    J  37378000.0   
32  C1749     848    아파트  경기도   행복주택  36.70       200  34.0    J  37378000.0   
33  C1749     848    아파트  경기도   행복주택  36.70        24  34.0    J   7378000.0   
34  C1749     848    아파트  경기도   행복주택  36.70        24  34.0    J   7378000.0   
35  C1749     848    아파트  경기도   행복주택  36.70        24  34.0    J   7378000.0   
36  C1749     848    아파트  경기도   행복주택  36.70        24  34.0    J   7378000.0   
37  C1749     848    아파트  경기도   행복주택  36.70        24  34.0    J   7378000.0   
38  C1749     848    아파트  경기도   행복주택  36.70        24  34.0    J   7378000.0   

    ...  시군구코드     시군구명              단지명                     도로명주소 총세대수_y  \
0   ...    210      광명시          경기도 광명시     경기도 광명시 오리로994번길 10-1      8   
1   ...    220      평택시   평택소사벌 A-6 행복주택           경기도 평택시 죽백4로 60    840   
2   ...    210      광명시          경기도 광명시     경기도 광명시 오리로994번길 10-1      8   
3   ...    220      평택시   평택소사벌 A-6 행복주택           경기도 평택시 죽백4로 60    840   
4   ...    117  수원시 영통구          경기도 수원시  경기도 수원시 영통구 영통로345번길 1-7     16   
5   ...    117  수원시 영통구          경기도 수원시  경기도 수원시 영통구 영통로345번길 1-7     16   
6   ...    220      평택시   평택소사벌 A-6 행복주택           경기도 평택시 죽백4로 60    840   
7   ...    220      평택시   평택소사벌 A-6 행복주택           경기도 평택시 죽백4로 60    840   
8   ...    390      시흥시  시흥목감 A-5BL 13단지      경기도 시흥시 목감둘레로 229-15    240   
9   ...    390      시흥시  시흥목감 A-5B

- C1749: 평택소사벌 A-6 행복주택 / 경기도 평택시 죽백4로 60	

In [96]:
pd.merge(test_아파트[test_아파트.단지코드=="C1414"], data, on=["지역", "임대료","전용면적"], how='inner')

단지코드  총세대수_x 임대건물구분   지역 공급유형_x   전용면적  전용면적별세대수   공가수 자격유형     임대보증금_x  \
0  C1414     873    아파트  경기도   행복주택  16.80       338  29.0    J  16520000.0   
1  C1414     873    아파트  경기도   행복주택  16.80        63  29.0    J  16520000.0   
2  C1414     873    아파트  경기도   행복주택  26.65        93  29.0    J  24840000.0   
3  C1414     873    아파트  경기도   행복주택  26.82        18  29.0    J  25000000.0   
4  C1414     873    아파트  경기도   행복주택  36.56        44  29.0    J  32920000.0   

   ...   기관명  시군구코드  시군구명             단지명                     도로명주소  총세대수_y  \
0  ...  LH서울    360  남양주시  남양주장현 행복주택 2단지  경기도 남양주시 진접읍 금강로 1530-33     870   
1  ...  LH서울    360  남양주시  남양주장현 행복주택 2단지  경기도 남양주시 진접읍 금강로 1530-33     870   
2  ...  LH서울    360  남양주시  남양주장현 행복주택 2단지  경기도 남양주시 진접읍 금강로 1530-33     870   
3  ...  LH서울    360  남양주시  남양주장현 행복주택 2단지  경기도 남양주시 진접읍 금강로 1530-33     870   
4  ...  LH서울    360  남양주시  남양주장현 행복주택 2단지  경기도 남양주시 진접읍 금강로 1530-33     870   

  공급유형_y 주택유형명 단지내주차면수_y   임대보증금_y  
0   행복주택   아파트         0  18400000  
1   행복주택   아파트         0  18400000  
2   행복주택   아파트         0  27200000  
3   행복주택   아파트         0  27200000  
4   행복주택   아파트         0  36000000  

[5 rows x 24 columns]

- C1414: 남양주장현 행복주택 2단지 / 경기도 남양주시 진접읍 금강로 1530-33

In [97]:
pd.merge(test_아파트[test_아파트.단지코드=="C1683"], data, on=["지역", "전용면적", "임대보증금"], how='inner')

단지코드  총세대수_x 임대건물구분   지역 공급유형_x   전용면적  전용면적별세대수   공가수 자격유형       임대보증금  \
0  C1683     328    아파트  경기도   행복주택  21.54        40  14.0    J  14498000.0   
1  C1683     328    아파트  경기도   행복주택  21.54        12  14.0    J  14498000.0   
2  C1683     328    아파트  경기도   행복주택  21.54        64  14.0    J  14498000.0   
3  C1683     328    아파트  경기도   행복주택  21.54        16  14.0    J  14498000.0   
4  C1683     328    아파트  경기도   행복주택  26.41        56  14.0    J  17434000.0   
5  C1683     328    아파트  경기도   행복주택  26.41         4  14.0    J  17434000.0   
6  C1683     328    아파트  경기도   행복주택  36.64        69  14.0    J  24186000.0   
7  C1683     328    아파트  경기도   행복주택  36.64        65  14.0    J  24186000.0   

   ...   기관명  시군구코드  시군구명              단지명                 도로명주소  총세대수_y  \
0  ...  LH경기    220   평택시  평택청북 B12BL 행복주택  경기도 평택시 청북읍 안청로1길 21     326   
1  ...  LH경기    220   평택시  평택청북 B12BL 행복주택  경기도 평택시 청북읍 안청로1길 21     326   
2  ...  LH경기    220   평택시  평택청북 B12BL 행복주택  경기도 평택시 청북읍 안청로1길 21     326   
3  ...  LH경기    220   평택시  평택청북 B12BL 행복주택  경기도 평택시 청북읍 안청로1길 21     326   
4  ...  LH경기    220   평택시  평택청북 B12BL 행복주택  경기도 평택시 청북읍 안청로1길 21     326   
5  ...  LH경기    220   평택시  평택청북 B12BL 행복주택  경기도 평택시 청북읍 안청로1길 21     326   
6  ...  LH경기    220   평택시  평택청북 B12BL 행복주택  경기도 평택시 청북읍 안청로1길 21     326   
7  ...  LH경기    220   평택시  평택청북 B12BL 행복주택  경기도 평택시 청북읍 안청로1길 21     326   

  공급유형_y 주택유형명 단지내주차면수_y   임대료_y  
0   행복주택   아파트         0   65210  
1   행복주택   아파트         0   65210  
2   행복주택   아파트         0   65210  
3   행복주택   아파트         0   65210  
4   행복주택   아파트         0   81700  
5   행복주택   아파트         0   81700  
6   행복주택   아파트         0  113360  
7   행복주택   아파트         0  113360  

[8 rows x 24 columns]

- C1683: 평택청북 B12BL 행복주택/경기도 평택시 청북읍 안청로1길 21

In [98]:
pd.merge(test_아파트[test_아파트.단지코드=="C1038"], data, on=["지역", "공급유형", "전용면적"], how='inner')

단지코드  총세대수_x 임대건물구분    지역  공급유형   전용면적  전용면적별세대수   공가수 자격유형     임대보증금_x  \
0  C1038     554    아파트  충청북도  행복주택  16.94       136  34.0    K  35897000.0   
1  C1038     554    아파트  충청북도  행복주택  26.56       140  34.0    K  57678000.0   
2  C1038     554    아파트  충청북도  행복주택  26.56        36  34.0    K  57060000.0   
3  C1038     554    아파트  충청북도  행복주택  26.76        12  34.0    K  57060000.0   
4  C1038     554    아파트  충청북도  행복주택  36.76       208  34.0    K  79833000.0   
5  C1038     554    아파트  충청북도  행복주택  36.95        18  34.0    K  79833000.0   

   ...   기관명  시군구코드  시군구명             단지명               도로명주소  총세대수_y 주택유형명  \
0  ...  LH충북    130   충주시  충주호암A4블록(행복주택)  충청북도 충주시 호암수청1로 30     550   아파트   
1  ...  LH충북    130   충주시  충주호암A4블록(행복주택)  충청북도 충주시 호암수청1로 30     550   아파트   
2  ...  LH충북    130   충주시  충주호암A4블록(행복주택)  충청북도 충주시 호암수청1로 30     550   아파트   
3  ...  LH충북    130   충주시  충주호암A4블록(행복주택)  충청북도 충주시 호암수청1로 30     550   아파트   
4  ...  LH충북    130   충주시  충주호암A4블록(행복주택)  충청북도 충주시 호암수청1로 30     550   아파트   
5  ...  LH충북    130   충주시  충주호암A4블록(행복주택)  충청북도 충주시 호암수청1로 30     550   아파트   

  단지내주차면수_y   임대보증금_y   임대료_y  
0         0  11600000   69310  
1         0  18640000  111370  
2         0  18640000  111370  
3         0  18440000  110170  
4         0  25800000  154150  
5         0  25800000  154150  

[6 rows x 24 columns]

- C2497: 경북혁신도시 Ab9블록 / 경상북도 김천시 용전3로 10
- C1749: 평택소사벌 A-6 행복주택 / 경기도 평택시 죽백4로 60
- C1414: 남양주장현 행복주택 2단지 / 경기도 남양주시 진접읍 금강로 1530-33
- C1683: 평택청북 B12BL 행복주택/경기도 평택시 청북읍 안청로1길 21
- C1038: 충주호암A4블록(행복주택)/ 충청북도 충주시 호암수청1로 30

In [99]:
indata_test6=pd.DataFrame({'단지코드':['C2497', 'C1749', 'C1414', 'C1683', 'C1038'],
                     '단지명':['경북혁신도시 Ab9블록', 
                            '평택소사벌 A-6 행복주택',
                            '남양주장현 행복주택 2단지',
                            '평택청북 B12BL 행복주택',
                            '충주호암A4블록(행복주택)'],
                     '도로명주소':['경상북도 김천시 용전3로 10',
                              '경기도 평택시 죽백4로 60', 
                              '경기도 남양주시 진접읍 금강로 1530-33',
                              '경기도 평택시 청북읍 안청로1길 21',
                              '충청북도 충주시 호암수청1로 30']})

In [100]:
indata_test6

단지코드              단지명                     도로명주소
0  C2497     경북혁신도시 Ab9블록          경상북도 김천시 용전3로 10
1  C1749   평택소사벌 A-6 행복주택           경기도 평택시 죽백4로 60
2  C1414   남양주장현 행복주택 2단지  경기도 남양주시 진접읍 금강로 1530-33
3  C1683  평택청북 B12BL 행복주택      경기도 평택시 청북읍 안청로1길 21
4  C1038   충주호암A4블록(행복주택)        충청북도 충주시 호암수청1로 30

In [101]:
df=pd.concat([test_store, indata_test, indata_test2, indata_test3, indata_test4, indata_test5, indata_test6], axis=0).reset_index(drop=True)

In [102]:
df[df.duplicated('단지코드')]

단지코드                        단지명                     도로명주소
13   C1006                      대전둔산1          대전광역시 서구 월평북로 11
58   C2675  흥덕마을휴먼시아1단지아파트(101동~109동)    경기도 용인시 기흥구 흥덕1로79번길 7
92   C2033                   달동 주공3단지       울산광역시 남구 번영로107번길 5
93   C1812                     울산화정주공         울산광역시 동구 월봉10길 16
96   C1729                   춘천효자8 단지        강원도 춘천시 공지로 234-16
103  C1887                  대구다사주공아파트   대구광역시 달성군 다사읍 매곡로14길 13
136  C2676                    모라주공3단지  부산광역시 사상구 모라로192번길 20-21
137  C2177                  읍내동 주공아파트         충청남도 아산시 외암로 1420
138  C2253                  명륜주공2차아파트           강원도 원주시 예술관길 31

In [103]:
df[df.duplicated()]

단지코드        단지명                     도로명주소
13   C1006      대전둔산1          대전광역시 서구 월평북로 11
92   C2033   달동 주공3단지       울산광역시 남구 번영로107번길 5
93   C1812     울산화정주공         울산광역시 동구 월봉10길 16
136  C2676    모라주공3단지  부산광역시 사상구 모라로192번길 20-21
137  C2177  읍내동 주공아파트         충청남도 아산시 외암로 1420
138  C2253  명륜주공2차아파트           강원도 원주시 예술관길 31

In [104]:
df= df.drop_duplicates(keep='first').reset_index(drop=True)

In [105]:
df[df.duplicated('단지코드')]

단지코드                        단지명                    도로명주소
57   C2675  흥덕마을휴먼시아1단지아파트(101동~109동)   경기도 용인시 기흥구 흥덕1로79번길 7
93   C1729                   춘천효자8 단지       강원도 춘천시 공지로 234-16
100  C1887                  대구다사주공아파트  대구광역시 달성군 다사읍 매곡로14길 13

In [106]:
df[df.단지코드=='C2675']

단지코드                        단지명                   도로명주소
56  C2675  흥덕마을휴먼시아1단지아파트(110동~115동)  경기도 용인시 기흥구 흥덕1로79번길 7
57  C2675  흥덕마을휴먼시아1단지아파트(101동~109동)  경기도 용인시 기흥구 흥덕1로79번길 7

In [107]:
df[df.단지코드=='C1729'] # 놀랍게도 후자가 맞음

단지코드       단지명               도로명주소
6   C1729   춘천효자8단지  강원도 춘천시 공지로 234-16
93  C1729  춘천효자8 단지  강원도 춘천시 공지로 234-16

In [108]:
df[df.단지코드=='C1887']

단지코드        단지명                    도로명주소
99   C1887  대구다사주공아파트  대구광역시 달성군 다사읍 매곡로14길 11
100  C1887  대구다사주공아파트  대구광역시 달성군 다사읍 매곡로14길 13

In [110]:
df.shape # 그래서 3개가 더 있는 것이다!

(153, 3)

# done!

In [112]:
df.to_excel("test_result.xlsx", index=False)